discord/app_commands/commands.py

```diff
@@ -618,6 +622,9 @@ class Command(Generic[GroupT, P, T]):
         Whether the command should only be usable in guild contexts.
 
         Due to a Discord limitation, this does not work on subcommands.
+    restrict_contexts: Optional[List[:class:`~discord.AppCommandContext`]]
```

Comment:
Originally I meant that the attribute should just be `restrict` or `restrictions` but I think now `restrict` would make more sense as the decorator that applies multiple if one would ever happen and the attribute name itself should just be reverted back to `allowed_contexts`.

Sorry for the churn. Bikeshedding tends to be hard :)

```suggestion
    allowed_contexts: Optional[:class:`~discord.AppCommandContext`]
```

Note that since this is a flag now the `List[...]` is no longer applicable.

Reply:
Nah, those names are fine. I was originally thinking of a new decorator that does e.g. `@restrict(guilds=True, private_channel=False)` but due to the inconsistency in the way the thing was set up (no group DM only nonsense) this idea ended up being kind of mediocre so no point pursuing it.